### PR TITLE
12.0 mig base usability

### PR DESCRIPTION
**Pull request by rvalyi**
Merry Christmas Akretion!

This is @alexis-via base_usability migrated to v12. I think I tested all features (though the module has no test and I didn't add any). Well, I droped the security CSV rules for country, state, and partner title as now there is a new group called 'Extra Rights / Contact Creation' which manages these rights. I also moved the formatLang method into the misc.py file and re-organised the files according to the OCA layout. Also partners and companies have no more fax field in Odoo 12 so I removed it from the reporting methods.

cc @renatonlima @mbcosta I think we can use this module without problem in Brazil. Eventually we can propose some changes in the reporting methods for partner and company.

### CODE REVIEW

File: base_usability/models/partner.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+from odoo import models, fields, api, _
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    # track_visibility is handled in the 'mail' module, and base_usability
+    # doesn't depend on 'mail', but that doesn't hurt, it will just be
+    # ignored if mail is not installed
+    name = fields.Char(track_visibility='onchange')
```
Comment (bealdav) [inline base_usability/models/partner.py:13]: Track_visibility becomes boolean attribute in v12

**Review by bealdav — CHANGES_REQUESTED** [review]

File: base_usability/models/partner.py (hunk 1 above)
Comment (rvalyi) [reply base_usability/models/partner.py:13]: @bealdav where from do you get this? I see the same kind of values instead, 'onchange' or 'always', see for instance:
https://github.com/odoo/odoo/blob/12.0/addons/account/models/account_invoice.py#L244
https://github.com/odoo/odoo/blob/12.0/addons/account/models/account_invoice.py#L262

Comment (bealdav) [reply base_usability/models/partner.py:13]: Oh, sorry for the fake news. It was in master

https://github.com/odoo/odoo/blob/master/addons/mail/models/mail_blacklist.py#L20

File: base_usability/fix-odoo-base-multi-company-partners-not-shared.diff [hunk 2]
```diff
@@ -0,0 +1,18 @@
+diff --git a/odoo/addons/base/res/res_users.py b/odoo/addons/base/res/res_users.py
+index c621ca4..d3880fe 100644
+--- a/odoo/addons/base/res/res_users.py
++++ b/odoo/addons/base/res/res_users.py
+@@ -334,7 +334,12 @@ class Users(models.Model):
+         user = super(Users, self).create(vals)
```
Comment (bealdav) [inline base_usability/fix-odoo-base-multi-company-partners-not-shared.diff]: The code here has been changed in v12 to create_multi.
https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/res_users.py#L423-L429

But i don't use this patch, then not a problem for me

**Review by bealdav — APPROVED** [review]

File: base_usability/fix-odoo-base-multi-company-partners-not-shared.diff (hunk 2 above)
Comment (rvalyi) [reply base_usability/fix-odoo-base-multi-company-partners-not-shared.diff]: I updated this patch in my last commit anyway

**Pull request merged**